src/Gate.java

```diff
@@ -104,31 +108,30 @@ public void main(String[] args) throws MsgException {
 					//   2) Gate release
 					//   3) workflow specific parameters
 					// If less than 3 files were found in the catalog, exit.
-					// TODO replace this by a loop
 					downloadTime.start();
 					if (VIPSimulator.gateInputFileNames.size() < 3){
 						Msg.error("Some input files are missing. Exit!");
 						System.exit(1);
 					}
-					transfer_info = LCG.cp(
-							VIPSimulator.gateInputFileNames.get(0),
-							"/scratch/gate.sh.tar.gz",
-							VIPServer.getDefaultLFC());
-					System.err.println(jobId + "," + getHost().getName() + ","
-							+ transfer_info + ",2");
-
-					transfer_info = LCG.cp(
-							VIPSimulator.gateInputFileNames.get(1),
-							"/scratch/gate_release.tar.gz",
-							VIPServer.getDefaultLFC());
-					System.err.println(jobId + "," + getHost().getName() + ","
-							+ transfer_info + ",2");
-
-					transfer_info = LCG.cp(
-							VIPSimulator.gateInputFileNames.get(2),
-							"/scratch/file.zip", VIPServer.getDefaultLFC());
-					System.err.println(jobId + "," + getHost().getName() + ","
-							+ transfer_info + ",2");
+					Vector<SE> replicaLocations = new Vector<SE>();
+					String logicalFileName;
+					
+					for(int i=0;i< VIPSimulator.gateInputFileNames.size();i++){
```

Comment:
Replace by (untested)
Iterator iLocalNames = VIPSimulator.gateInputLocalNames.iterator();
for (String LogicalFileName : VIPSimulator.gateInputFileNames){
    localFileName = iLocalNames.next() // that replaces localOuput[i]